### PR TITLE
Open ai

### PR DESCRIPTION
**Pull request by alifarooq96**
## Description

Please include a summary of the changes and issues addressed in this PR. Also, explain the motivation and context of each update. List any dependencies or prerequisites needed for implementing this change.

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/open-ai/src/app/install/service.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  region: string;
+};
+
+export const registerOrganisation = async ({
+  organisationId,
+  token,
+  sourceOrganizationId,
+  region,
+}: SetupOrganisationParams) => {
+  const [organisation] = await db
```
Comment (elbalexandre) [inline apps/open-ai/src/app/install/service.ts:18]: We need to do a check to see if organizationId & token given by user is valid.
As their is no API endpoints designed for this, we can fetch user here.
```suggestion
  // api call to check that token & sourceOrganizaton are correct
  await getUsers(token, sourceOrganizationId)

  const [organisation] = await db
```

File: apps/open-ai/src/app/install/action.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    };
+  }
+
+  const organisationId = result.data.organisationId;
+  const region = result.data.region;
+  const token = result.data.token;
+  const sourceOrganizationId = result.data.sourceOrganizationId;
+
+  try {
+    await registerOrganisation({ organisationId, token, sourceOrganizationId, region });
```
Comment (elbalexandre) [inline apps/open-ai/src/app/install/action.tsx:50]:
```suggestion
  try {
    await registerOrganisation(result.data);
```

File: apps/open-ai/src/database/client.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { drizzle } from 'drizzle-orm/postgres-js';
+import postgres from 'postgres';
+import { env } from '@/env';
+import * as schema from './schema';
+
+const sql = postgres(env.DATABASE_URL);
```
Comment (elbalexandre) [inline apps/open-ai/src/database/client.ts:6]: It's the wrong client file, it's not edge-runtime compatible (you can replace it with client.ts from template)

Comment (awan7676) [reply apps/open-ai/src/database/client.ts]: Done!

File: apps/open-ai/src/app/install/action.tsx (hunk 2 above)
Comment (awan7676) [reply apps/open-ai/src/app/install/action.tsx]: Done!

File: apps/open-ai/src/app/install/service.ts (hunk 1 above)
Comment (awan7676) [reply apps/open-ai/src/app/install/service.ts:20]: Done!

File: apps/open-ai/src/connectors/delete-user.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { OpenAiError } from './commons/error';
```
[the remark this answers is not in the document]
Comment (awan7676) [reply apps/open-ai/src/connectors/delete-user.ts]: Done!

File: apps/open-ai/src/app/install/service.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  region: string;
+};
+
+export const registerOrganisation = async ({
+  organisationId,
+  token,
+  sourceOrganizationId,
+  region,
+}: SetupOrganisationParams) => {
+  await getUsers(token, sourceOrganizationId);
```
Comment (elbalexandre) [inline apps/open-ai/src/app/install/service.ts:19]:
```suggestion
  // api call to check that token & sourceOrganizaton are correct
  await getUsers(token, sourceOrganizationId);
```

**Review by elbalexandre — APPROVED** [review]

**Pull request merged**